applications/asset_tracker_v2/tests/lwm2m_integration/src/lwm2m_integration_test.c

```diff
@@ -0,0 +1,344 @@
+						     endpoint_name,
+						     0,
+						     NULL,
+						     NULL,
+						     0);
+	__wrap_lwm2m_rd_client_start_IgnoreArg_event_cb();
+
+	TEST_ASSERT_EQUAL(cloud_wrap_connect(), 0);
+
+	/* Trigger internal events that puts the uut in CONNECTED state. */
+	rd_client_callback(&client, LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_COMPLETE);
+	rd_client_callback(&client, LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE);
+}
+
+void test_lwm2m_integration_disconnect(void)
+{
+	test_lwm2m_integration_connect();
+
+	__wrap_lwm2m_rd_client_stop_ExpectAndReturn(&client, NULL, false, 0);
+	__wrap_lwm2m_rd_client_stop_IgnoreArg_event_cb();
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_disconnect());
+}
+
+void test_lwm2m_integration_data_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_data_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+void test_lwm2m_integration_ui_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_ui_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+void test_lwm2m_integration_neighbor_cells_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_neighbor_cells_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+void test_lwm2m_integration_agps_request_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_agps_request_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+/* Tests for unsupported APIs. */
+void test_lwm2m_integration_state_get(void)
+{
+	TEST_ASSERT_EQUAL(-ENOTSUP, cloud_wrap_state_get(true, 0));
+}
+
+void test_lwm2m_integration_state_send(void)
+{
+	TEST_ASSERT_EQUAL(-ENOTSUP, cloud_wrap_state_send(NULL, 0, true, 0));
+}
+
+void test_lwm2m_integration_batch_send(void)
+{
+	TEST_ASSERT_EQUAL(-ENOTSUP, cloud_wrap_batch_send(NULL, 0, true, 0));
+}
+
+void test_lwm2m_integration_pgps_request_send(void)
+{
+	TEST_ASSERT_EQUAL(-ENOTSUP, cloud_wrap_pgps_request_send(NULL, 0, true, 0));
+}
+
+void test_lwm2m_integration_memfault_data_send(void)
+{
+	TEST_ASSERT_EQUAL(-ENOTSUP, cloud_wrap_memfault_data_send(NULL, 0, true, 0));
+}
+
+void test_lwm2m_integration_reboot_request(void)
```

Comment:
I find it hard to understand what this test does. I believe most of the testing is done in the setup function which is not visible here. Can you add a line or two that explains what this does? 

Also since write in line 259 that these are tests for unsupported APIs, it confused me a bit. This test and every test below is not a test for unsupported APIs.

Reply:
I think a one liner explanation for each of the tests below do would help a lot.  :-)

Reply:
Updated the test file with some more hopefully descriptive comments.